### PR TITLE
introduce nonClampedIntArrayConstructors

### PR DESCRIPTION
**Pull request by bakkot**
Per https://github.com/tc39/test262/pull/3849/files/e5d20a7fb8ed1beeeba3eb75c21aed433de164ac#r1447792194, oops.

### CODE REVIEW

**Review by ljharb — APPROVED** [review]

**Comment by bakkot** [comment]
Changed this so that instead `intArrayConstructors` includes `Uint8ClampedArray`, and there's a new `nonClampedIntArrayConstructors` used in the tests which need it per [this comment](https://github.com/tc39/test262/pull/3849#discussion_r1447841772).

(@ljharb's review was for the previous version of this PR, sorry.)

**Review by ljharb — APPROVED** [review]
I like this better anyways

File: harness/testTypedArray.js [hunk 1]
```diff
@@ -20,16 +20,17 @@ var floatArrayConstructors = [
   Float32Array
 ];
 
-var intArrayConstructors = [
+var nonClampedIntArrayConstructors = [
```
Comment (iamstolis) [inline harness/testTypedArray.js:23]: I guess that `nonClampedIntArrayConstructors` should be mentioned in the `defines` list above.

Comment (bakkot) [reply harness/testTypedArray.js:24]: Good catch, done.

**Review by Ms2ger — APPROVED** [review]

**Pull request merged**